### PR TITLE
[install UX] incrementally show the output in-line

### PR DESCRIPTION
**Pull request by savil**
## Summary

Thanks to @mikeland86's suggestion we can have snappy incremental in-line updates
of what the `nix-env --install` is doing.

## How was it tested?

added `buf` to a devbox.json and did `devbox shell`

### CODE REVIEW

**Comment by savil** [comment]
Current dependencies on/for this PR:
* main
  * **PR #506** <a href="https://app.graphite.dev/github/pr/jetpack-io/devbox/506" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jetpack-io/devbox/506?utm_source=stack-comment).

File: internal/impl/devbox.go [hunk 1]
```diff
@@ -726,7 +727,45 @@ func (d *Devbox) installNixProfile() (err error) {
 		}
 
 		cmd.Env = nix.DefaultEnv()
-		_, err = cmd.Output()
+
+		// Get a pipe to read from standard out
+		r, _ := cmd.StdoutPipe()
```
Comment (mikeland86) [inline internal/impl/devbox.go:732]: I think this is an error we probably want to handle?

File: internal/impl/devbox.go [hunk 2]
```diff
@@ -726,7 +727,45 @@ func (d *Devbox) installNixProfile() (err error) {
...
+
+		// Use the same pipe for standard error
+		cmd.Stderr = cmd.Stdout
```
Comment (mikeland86) [inline internal/impl/devbox.go:735]: This was from example I copy pasted (and probably correct) but just wanted to call out the previously we used `cmd.Output()` and not `CombinedOutput` so this changes it a bit.

File: internal/impl/devbox.go [hunk 3]
```diff
@@ -726,7 +727,45 @@ func (d *Devbox) installNixProfile() (err error) {
...
+				step.Display(fmt.Sprintf("%s   %s", msg, line))
+			}
+
+			// We're all done, unblock the channel
+			done <- struct{}{}
+		}()
+
+		// Start the command and check for errors
+		err := cmd.Start()
+		if err != nil {
```
Comment (mikeland86) [inline internal/impl/devbox.go:759]: nit same line

**Review by mikeland86 — APPROVED** [review]
Nice!

File: internal/impl/devbox.go (hunk 1 above)
Comment (savil) [reply internal/impl/devbox.go]: good call

File: internal/impl/devbox.go (hunk 2 above)
Comment (savil) [reply internal/impl/devbox.go]: true, but we were ignoring the output returned from `cmd.Output()`, so I think this is okay for now.

**Pull request merged**